lab_1_classify_by_unigrams/main.py

```diff
@@ -11,15 +11,29 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
 
+    return [i for i in text.lower() if i.isalpha()]
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list) or not all([isinstance(i, str) for i in tokens]):
+        return None
 
+    tokens_count = len(tokens)
+    list_tokens = {}
+    for i in tokens:
+        if i in list_tokens:
+            list_tokens[i] += 1
+        else:
+            list_tokens[i] = 1
```

Comment:
Let set 0 if i is not in tokens and always do += 1

Reply:
Done